### PR TITLE
142 fix wordclouds erase mechanic

### PR DESCRIPTION
**Pull request by nano1342**
Made it so the text that's stored in the front and in the back are more synchronized. If the player pastes a text in the field, presses ctrl + backspace or types in other exotic ways, the requests should correspond to the actual modifications.
Though, its spams the requests a lot, so I've disabled pasting, text selection and ctrl z to prevent sending too many requests. Selecting is still possible if the user is fast enough.

Also, I realized the word isn't synchronized if the player refreshes their page. It seems hard to fix for new so we'll have to discuss this later.

### CODE REVIEW

**Comment by Picorims** [comment]
> Also, I realized the word isn't synchronized if the player refreshes their page. It seems hard to fix for new so we'll have to discuss this later.

Extending the current game state server event would allow adding this context to the front (what is stored as typed by the user). It could serve to prefill the input. But that's a separate issue: #149

File: src-front/src/components/WordCloud/WordCloud.tsx [hunk 1]
```diff
@@ -154,17 +154,38 @@ export default function WordCloud({
   const [inputValue, setInputValue] = useState("");
   //const [error, setError] = useState("");
   const [scores, setScores] = useState<Record<string, number>>({});
-  const [wordsBestProgress, setWordsBestProgress] = useState<
-    Record<string, number>
-  >({});
+  // const [wordsBestProgress, setWordsBestProgress] = useState<
```
Comment (Picorims) [inline src-front/src/components/WordCloud/WordCloud.tsx:157]: Is support for displaying global best progress dropped?

File: src-front/src/components/WordCloud/WordCloud.tsx [hunk 2]
```diff
@@ -154,17 +154,38 @@ export default function WordCloud({
...
 
   //WordCloud initialization
   const init = useCallback(async () => {
+    //forbids pasting text
+    const playerInput = document.getElementById(
+      "playerInput",
+    ) as HTMLInputElement;
+    playerInput.onpaste = (e) => e.preventDefault();
+    //forbids text selection
+    playerInput.addEventListener(
```
Comment (Picorims) [inline src-front/src/components/WordCloud/WordCloud.tsx:175]: It needs an equivalent remove event listener for each event listener created.

**Review by Picorims — CHANGES_REQUESTED** [review]

File: src-front/src/components/WordCloud/WordCloud.tsx (hunk 1 above)
Comment (nano1342) [reply src-front/src/components/WordCloud/WordCloud.tsx:157]: Idk I thought I wasn't coded in the back, wordsBestProgress is always empty on my side

File: src-front/src/components/WordCloud/WordCloud.tsx (hunk 2 above)
Comment (nano1342) [reply src-front/src/components/WordCloud/WordCloud.tsx]: I just removed it, it wasn't a good idea anyway

File: src-front/src/components/WordCloud/WordCloud.tsx (hunk 1 above)
Comment (Picorims) [reply src-front/src/components/WordCloud/WordCloud.tsx:157]: It is supposed to be fully functional, if not it is a bug.

File: src-front/src/components/WordCloud/WordCloud.tsx (hunk 2 above)
Comment (Picorims) [reply src-front/src/components/WordCloud/WordCloud.tsx]: But I mean, for all events.

**Review by Picorims — APPROVED** [review]

File: src-front/src/components/WordCloud/WordCloud.tsx (hunk 1 above)
Comment (Picorims) [reply src-front/src/components/WordCloud/WordCloud.tsx:157]: see: #154

**Pull request merged**